src/main/java/demohometask/demohometask/Match.java

```diff
@@ -0,0 +1,92 @@
+package demohometask.demohometask;
+
+import org.springframework.stereotype.Component;
+
+import java.io.FileOutputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+
+@Component
+public class Match {
+    private static int round;
```

Comment:
1) don't use static members in Spring component;
2) if you declare your spring bean as singleton(default behaviour) you shouldn't mutate it's state. Consider moving current round into Tournament object that you'll store on the filesystem and that will represent current state of the tournament